cms/djangoapps/contentstore/views/helpers.py

```diff
@@ -197,6 +197,13 @@ def create_xblock(parent_locator, user, category, display_name, boilerplate=None
 
         if display_name is not None:
             metadata['display_name'] = display_name
+        elif 'display_name' in metadata:
```

Comment:
This would probably be a good place to put in comments - put comments before each if/elif/else statement to explain what each block is doing.